invokeai/app/api/routers/model_manager.py

```diff
@@ -816,3 +814,79 @@ async def get_starter_models() -> list[StarterModel]:
         model.dependencies = missing_deps
 
     return starter_models
+
+
+@model_manager_router.get(
+    "/model_cache",
+    operation_id="get_cache_size",
+    response_model=float,
+    summary="Get maximum size of model manager RAM or VRAM cache.",
+)
+async def get_cache_size(cache_type: CacheType = Query(description="The cache type", default=CacheType.RAM)) -> float:
+    """Return the current RAM or VRAM cache size setting (in GB)."""
+    cache = ApiDependencies.invoker.services.model_manager.load.ram_cache
+    if cache_type == CacheType.RAM:
+        return cache.max_cache_size
+    elif cache_type == CacheType.VRAM:
+        return cache.max_vram_cache_size
+    else:
+        raise ValueError(f"Unexpected {cache_type=}.")
+
+
+@model_manager_router.put(
+    "/model_cache",
+    operation_id="set_cache_size",
+    response_model=float,
+    summary="Set maximum size of model manager RAM or VRAM cache, optionally writing new value out to invokeai.yaml config file.",
+)
+async def set_cache_size(
+    value: float = Query(description="The new value for the maximum cache size"),
+    cache_type: CacheType = Query(description="The cache type", default=CacheType.RAM),
+    persist: bool = Query(description="Write new value out to invokeai.yaml", default=False),
+) -> float:
+    """Set the current RAM or VRAM cache size setting (in GB). ."""
+    cache = ApiDependencies.invoker.services.model_manager.load.ram_cache
+    app_config = get_config()
+    vram_bak, ram_bak = (app_config.vram, app_config.ram)
+
+    if cache_type == CacheType.RAM:
+        cache.max_cache_size = value
+        app_config.ram = value
+    elif cache_type == CacheType.VRAM:
+        cache.max_vram_cache_size = value
+        app_config.vram = value
+    else:
+        raise ValueError(f"Unexpected {cache_type=}.")
+
+    if persist:
+        config_path = app_config.config_file_path
+        new_config_path = config_path.with_suffix(".yaml.new")
+        backup_config_path = config_path.with_suffix(".yaml.bak")
+        shutil.copy(config_path, backup_config_path)
+        try:
+            app_config.write_file(new_config_path)
+            shutil.move(new_config_path, config_path)
+        except Exception as e:
+            shutil.move(backup_config_path, config_path)
+            app_config.max_vram_cache_size = vram_bak
+            app_config.max_cache_size = ram_bak
+            raise RuntimeError(f"Failed to save configuration to {config_path}: {e}") from e
+
+    if cache_type == CacheType.VRAM:
+        return cache.max_vram_cache_size
+    elif cache_type == CacheType.RAM:
+        return cache.max_cache_size
+    else:
+        raise ValueError(f"Unexpected {cache_type=}.")
```

Comment:
I think we may have made this more complicated without really making it more robust. How about this (not tested):
```suggestion
    # Record initial state.
    vram_old = app_config.vram
    ram_old = app_config.ram

    # Prepare target state.
    vram_new = vram_old
    ram_new = ram_old
    if cache_type == CacheType.RAM:
        ram_new = value
    elif cache_type == CacheType.VRAM:
        vram_new = value
    else:
        raise ValueError(f"Unexpected {cache_type=}.")

    config_path = app_config.config_file_path
    new_config_path = config_path.with_suffix(".yaml.new")

    try:
        # Try to apply the target state.
        cache.max_vram_cache_size = vram_new
        cache.max_cache_size = ram_new
        app_config.max_cache_size = ram_new
        app_config.max_vram_cache_size = vram_new
        if persist:
            app_config.write_file(new_config_path)
            shutil.move(new_config_path, config_path)
    except Exception as e:
        # If there was a failure, restore the initial state.
        cache.max_vram_cache_size = vram_old
        cache.max_cache_size = ram_old
        app_config.max_cache_size = ram_old
        app_config.max_vram_cache_size = vram_old

        raise RuntimeError("Failed to update cache size") from e
```

Previously, we were relying on shutil.move(...) to do the cleanup in case of error. If we trust  `shutil.move()`, then we may as well just rely on it to perform the change and report success or failure.